### PR TITLE
feat: fix 404 page; redirect undefined/unfindable searches to 404; redirect Internal Server Error to 404;

### PR DESCRIPTION
**Pull request by dev4pear**

### CODE REVIEW

File: routes/404.tsx [hunk 1]
```diff
@@ -9,28 +9,22 @@ export default function Error404() {
       <div class="px-4 py-8 mx-auto">
         <div class="max-w-screen-md mx-auto flex flex-col items-center justify-center">
           <div class="flex flex-col justify-center items-center">
-            <div class="inline-block text-3xl mobileMd:text-4xl mobileLg:text-5xl desktop:text-6xl font-black purple-gradient1">
```
[the remark this answers is not in the document]
Comment (reinamora137) [reply routes/404.tsx:12]: this was changed again in PR https://github.com/stampchain-io/BTCStampsExplorer/pull/422

so these modifications can be ignored here

File: routes/collection/details/[id].tsx [hunk 2]
```diff
@@ -77,10 +77,13 @@ export const handler: Handlers = {
       return await ctx.render(data);
     } catch (error) {
       console.error("Error in collection details:", error);
-      if (error.message?.includes("Collection not found")) {
-        return ctx.renderNotFound();
-      }
-      return new Response("Internal Server Error", { status: 500 });
+      return new Response("", {
```
Comment (reinamora137) [inline routes/collection/details/[id].tsx:80]: this should use ResponseUtil to return the 500 via internalError

File: routes/home/index.tsx [hunk 3]
```diff
@@ -79,10 +79,12 @@ export const handler: Handlers = {
       return ctx.render(data);
     } catch (error) {
       console.error("Error in handler:", error);
-      return ctx.render({
-        error: `Error: Internal server error`,
-        btcPrice: 0,
-        recommendedFee: 6,
+      return new Response("", {
```
Comment (reinamora137) [inline routes/home/index.tsx:82]: this should use ResponseUtil internalError - should fall through to the 302 redirect to 404

**Comment by reinamora137** [comment]
Why was the _404.tsx removed? this appears to be the default route for fresh to fall through when you renderNoContent

**Comment by dev4pear** [comment]
i've restored _404.tsx and used ResponseUtil.internalError everywhere internal error comes out

File: routes/_404.tsx [hunk 4]
```diff
@@ -1,15 +1,6 @@
 import { Head } from "$fresh/runtime.ts";
 
 export default function Error404() {
-  const titleGreyLD =
```
[the remark this answers is not in the document]
Comment (reinamora137) [reply routes/_404.tsx:4]: the latest update changed these from gray to purple?

File: routes/_404.tsx [hunk 5]
```diff
@@ -18,26 +16,26 @@ export default function Error404() {
       <div class="px-4 py-8 mx-auto">
         <div class="max-w-screen-md mx-auto flex flex-col items-center justify-center">
           <div class="flex flex-col justify-center items-center">
-            <div class={titleGreyLD}>WHOOPS</div>
```
Comment (reinamora137) [inline routes/_404.tsx:21]: why was this changed from the prior updates babalicious made?

Comment (dev4pear) [reply routes/_404.tsx:21]: yes, it was changed gray to purple from the latest update
